server/sonar-server/src/main/java/org/sonar/server/setting/ws/SetAction.java

```diff
@@ -305,27 +312,26 @@ private static SetRequest toWsRequest(Request request) {
       .setKey(request.mandatoryParam(PARAM_KEY))
       .setValue(request.param(PARAM_VALUE))
       .setValues(request.multiParam(PARAM_VALUES))
-      .setFieldValues(readFieldValues(request))
+      .setFieldValues(request.multiParam(PARAM_FIELD_VALUES))
       .setComponentId(request.param(PARAM_COMPONENT_ID))
       .setComponentKey(request.param(PARAM_COMPONENT_KEY))
       .build();
   }
 
-  private static List<Map<String, String>> readFieldValues(Request request) {
-    String key = request.mandatoryParam(PARAM_KEY);
-
-    return request.multiParam(PARAM_FIELD_VALUES).stream()
-      .map(json -> readOneFieldValues(json, key))
-      .collect(Collectors.toList());
-  }
-
   private static Map<String, String> readOneFieldValues(String json, String key) {
     Type type = new TypeToken<Map<String, String>>() {
     }.getType();
+    Gson gson = GsonHelper.create();
     try {
-      return (Map<String, String>) GsonHelper.create().fromJson(json, type);
-    } catch (JsonParseException e) {
-      throw new BadRequestException(String.format("Invalid JSON '%s' for setting '%s'", json, key));
+      return (Map<String, String>) gson.fromJson(json, type);
+    } catch (JsonSyntaxException e) {
+      try {
+        gson.fromJson(json, Object.class);
+      } catch (JsonParseException invalidJson) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)